### PR TITLE
Fix certbot-auto on RHEL 8

### PR DESCRIPTION
**Pull request by bmw**
This PR builds off of #7240 to fix https://github.com/certbot/certbot/issues/7241.

The code in certbot-auto is unchanged which I +1. Someone else should give it a 2nd review.

For the code in the tests, you can see all tests passing (including `test_tests.sh`) at https://travis-ci.com/certbot/certbot/jobs/220681085.

One noteworthy thing here is I did not add the RHEL 8 AMI to the Apache tests due to https://github.com/certbot/certbot/issues/7273. This problem is not related to support in certbot-auto though, is an edge case, and I do not personally believe it should block this PR.

@adferrand, are you interested in giving this PR a review?

This PR should be merged and not squashed to preserve authorship.

### CODE REVIEW

File: letsencrypt-auto-source/letsencrypt-auto.template [hunk 1]
```diff
@@ -343,6 +343,14 @@ elif [ -f /etc/redhat-release ]; then
     }
     USE_PYTHON_3=1
     BOOTSTRAP_VERSION="BootstrapRpmPython3 $BOOTSTRAP_RPM_PYTHON3_VERSION"
+  # RHEL 8 uses python3.
+  elif [ "$RPM_DIST_NAME" = "rhel" -a "$RPM_DIST_VERSION" -ge 8 ]; then
```
Comment (adferrand) [inline letsencrypt-auto-source/letsencrypt-auto.template]: Is it possible to merge this condition with the previous one? It will reduce code duplication since the exact same logic is applied in case of old centos, rhel8 and fedora 29+.

Note that in my PR about migrating centos 6 users to python 3.6, I will extract this situation in a specific branch with a specific bootstrap.

File: tests/letstest/scripts/test_leauto_upgrades.sh [hunk 2]
```diff
@@ -26,6 +26,17 @@ else
     # 0.33.x is the oldest version of letsencrypt-auto that works on Fedora 29+.
     INITIAL_VERSION="0.33.1"
 fi
+
+# If we're on RHEL 8, the initial version of certbot-auto will fail until we do
```
Comment (adferrand) [inline tests/letstest/scripts/test_leauto_upgrades.sh]: We should create issue to mark the fact we need to remove this code once the next version of Certbot is released.

File: tests/letstest/scripts/test_sdists.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh -xe
 
 cd letsencrypt
-./certbot-auto --install-only -n --debug
+letsencrypt-auto-source/letsencrypt-auto --install-only -n --debug
```
Comment (adferrand) [inline tests/letstest/scripts/test_sdists.sh]: Why using the local newest letsencrypt-auto here instead of the released certbot-auto script? Does it mean the previous logic was wrong?

File: letsencrypt-auto-source/letsencrypt-auto.template (hunk 1 above)
Comment (bmw) [reply letsencrypt-auto-source/letsencrypt-auto.template]: I personally don't think we should make a massive line like:
```
if [ "$RPM_DIST_NAME" = "fedora" -a "$RPM_DIST_VERSION" -ge 29 -o "$RPM_DIST_NAME" = "rhel" -a "$RPM_DIST_VERSION" -ge 8 -o "$PYVER" -eq 26 ]; then
```
(plus any parens to make that work)

We could have separate if branches that set some variable like `RPM_USE_PYTHON_3` and then a separate if branch that checks this variable and then sets the relevant bootstrap variables.

If you think we should do this, I'd encourage you to suggest it on https://github.com/certbot/certbot/pull/7240 otherwise we need to get a 3rd person in here to review any changes we make.

If you don't like my suggestion to reuse code, I think we should get on the same page ourselves before making requests in #7240.

File: tests/letstest/scripts/test_sdists.sh (hunk 3 above)
Comment (bmw) [reply tests/letstest/scripts/test_sdists.sh]: We need to use letsencrypt-auto in letsencrypt-auto-source here because otherwise we fail to bootstrap and create a venv so we know whether to use Python 2 or Python 3 in this test script. We do the same kind of thing in `test_apache2.sh` and `test_test.sh`.

I wouldn't say the previous logic was necessarily wrong, just that it didn't allow the test to run on systems where bootstrapping with the latest released certbot-auto didn't work.

Comment (adferrand) [reply tests/letstest/scripts/test_sdists.sh]: OK it makes sense. And the new approach is more consistent, since the current letsencrypt-auto script becomes a part of the test, while we now that the released one is already working.

File: letsencrypt-auto-source/letsencrypt-auto.template (hunk 1 above)
Comment (adferrand) [reply letsencrypt-auto-source/letsencrypt-auto.template]: Yes my concern is to avoid duplication of the code, to improve readability and maintainability, but I agree that a big chunk of if condition does not reach this purpose.

I am OK with your proposition, it will make the code more explicit, while avoiding duplication of the code. 

I create a request on #7240.

File: tests/letstest/scripts/test_leauto_upgrades.sh (hunk 2 above)
Comment (bmw) [reply tests/letstest/scripts/test_leauto_upgrades.sh]: Created https://github.com/certbot/certbot/issues/7301.
